libraries/abstractions/pkcs11/CMakeLists.txt

```diff
@@ -23,27 +23,78 @@ afr_module_dependencies(
 )
 
 
-# Link to this INTERFACE target to use the default implementation based on mbedtls.
-afr_module(NAME pkcs11_mbedtls INTERFACE)
+if(ECC608_IN_USE)
+    set(mchp_dir "${AFR_VENDORS_DIR}/microchip")
+    set(ecc608a_dir "${mchp_dir}/secure_elements")
+    set(atca_config_dir "${CMAKE_CURRENT_LIST_DIR}/ecc608a")
+    afr_module(NAME pkcs11_ecc608a INTERFACE)
 
-set(src_dir "${CMAKE_CURRENT_LIST_DIR}/mbedtls")
+    afr_glob_src(ecc608a_atacert_src RECURSE DIRECTORY ${ecc608a_dir}/lib/atcacert)
+    afr_glob_src(ecc608a_basic_src RECURSE DIRECTORY ${ecc608a_dir}/lib/basic)
+    afr_glob_src(ecc608a_crypto_src RECURSE DIRECTORY ${ecc608a_dir}/lib/crypto)
+    afr_glob_src(ecc608a_host_src RECURSE DIRECTORY ${ecc608a_dir}/lib/host)
+    afr_glob_src(ecc608a_pkcs11_src RECURSE DIRECTORY ${ecc608a_dir}/lib/pkcs11)
 
-afr_module_sources(
-    pkcs11_mbedtls
-    INTERFACE
-        "${src_dir}/iot_pkcs11_mbedtls.c"
-        "${src_dir}/threading_alt.h"
-        "${inc_dir}/iot_pkcs11_pal.h"
-)
 
-afr_module_dependencies(
-    pkcs11_mbedtls
-    INTERFACE
-        AFR::crypto
-        AFR::pkcs11_implementation
-        AFR::utils
-        3rdparty::mbedtls
-)
+    afr_glob_src(ecc608a_src DIRECTORY ${ecc608a_dir}/lib)
+    list(REMOVE_ITEM ecc608a_src "${ecc608a_dir}/lib/atca_cfgs.c")
+
+    set(ecc608_include "${ecc608a_dir}/lib")
+    set(ecc608_src_list
+        "${ecc608a_src}"
+        "${ecc608a_atacert_src}"
+        "${ecc608a_basic_src}"
+        "${ecc608a_crypto_src}"
+        "${ecc608a_host_src}"
+        "${ecc608a_pkcs11_src}"
+    )
+
+    afr_module_sources(
+        ${AFR_CURRENT_MODULE}
```

Comment:
specify the module name as `pkcs11_ecc608a`  here to be consistent?

Reply:
Good catch, have updated now